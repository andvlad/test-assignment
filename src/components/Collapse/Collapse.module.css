.root {
  transition: max-height .3s ease-in-out;
  overflow-y: hidden;
}

.opened .content {
  margin-bottom: 4px;
}

.collapsed .content {
  max-height: 0;
}

.opened .content {
  max-height: 500px;
  transition: max-height .5s ease-in-out;
}

.label {
  display: flex;
  align-items: center;
  text-transform: uppercase;
  letter-spacing: 2px;
  font-size: 14px;
  cursor: pointer;
  padding: 4px 0 4px 0;
}

.label::before {
  content: '\02C4';
  display: inline-block;
  height: 23px;
  margin-right: 4px;
}

.collapsed .label::before {
  transform: rotateZ(180deg);
}

.opened .label::before {
  transition: transform .2s linear;
}
