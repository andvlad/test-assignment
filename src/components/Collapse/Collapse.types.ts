import React from 'react';

export interface ICollapse {
  label: string;
  className?: string;
  children: React.ReactNode;
}
