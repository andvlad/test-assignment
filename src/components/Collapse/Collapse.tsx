import React, {FormEvent, useCallback, useState} from 'react';

import { ICollapse } from './Collapse.types';
import styles from './Collapse.module.css';

export function Collapse({
  label,
  className,
  children,
}: ICollapse) {
  const [isCollapsed, setIsCollapsed] = useState<boolean>(true);

  const onClickCallback = useCallback((event: FormEvent<HTMLDivElement>) => {
    event.preventDefault();

    setIsCollapsed(!isCollapsed)
  }, [isCollapsed, setIsCollapsed])

  return (
    <div className={`${styles.root} ${isCollapsed ? styles.collapsed : styles.opened} ${className}`}>
      <div
        className={styles.label}
        onClick={onClickCallback}
      >
        {label}
      </div>
      <div className={styles.content}>
        {children}
      </div>
    </div>
  );
}
