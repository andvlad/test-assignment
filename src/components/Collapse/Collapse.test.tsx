import React from 'react';
import { render } from '@testing-library/react';

import { Collapse } from './Collapse';

it('matches snapshot', () => {
  const { asFragment } = render(
    <Collapse label="Payment">
      5000
    </Collapse>
  );

  expect(asFragment()).toMatchSnapshot();
});
