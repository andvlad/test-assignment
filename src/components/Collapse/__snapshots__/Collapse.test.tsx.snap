// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root collapsed undefined"
  >
    <div
      class="label"
    >
      Payment
    </div>
    <div
      class="content"
    >
      5000
    </div>
  </div>
</DocumentFragment>
`;
